src/inc/contract.h

```diff
@@ -2284,7 +2283,7 @@ class GetThreadAllowedHolder
 // also, we'd have contract bit for whether APIs can be called on the helper thread.
 // @todo - if we really wanted to be strict, we should make this per-thread.
 //-----------------------------------------------------------------------------
-#ifdef ENABLE_CONTRACTS_IMPL
+#ifdef FEATURE_INTEROP_DEBUGGING
```

Comment:
Should this be `#if defined(ENABLE_CONTRACTS_IMPL) && defined(FEATURE_INTEROP_DEBUGGING)` ?

We want to make sure that these things are not compiled into release build.

Reply:
That is not clear for me, so before this change we had definition of `g_DbgSuppressAllocationAsserts` under `ENABLE_CONTRACTS_IMPL` and uses under `DEBUG && USE_INTEROPSAFE_HEAP` and it worked because `USE_INTEROPSAFE_HEAP` is defined as `FEATURE_INTEROP_DEBUGGING` and it is defined as:
```
if (WIN32 AND (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_I386 OR CLR_CMAKE_TARGET_ARCH_ARM64))
    add_definitions(-DFEATURE_INTEROP_DEBUGGING)
endif (WIN32 AND (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_I386 OR CLR_CMAKE_TARGET_ARCH_ARM64))
```
so on all platforms where it was defined `ENABLE_CONTRACTS_IMPL` was also defined in debug.

Now it is different because for arm64 windows we want `ENABLE_CONTRACTS_IMPL` to be false and `USE_INTEROPSAFE_HEAP` to be true (Actually, do we?).

So if we change it to `#if defined(ENABLE_CONTRACTS_IMPL) && defined(FEATURE_INTEROP_DEBUGGING)` the use of `g_DbgSuppressAllocationAsserts` in  `F:\git\coreclr\src\debug\ee\rcthread.cpp::void AssertAllocationAllowed()` won't compile.

Reply:
So ifdef out that bad use to match

Reply:
@jkotas Fixed.

Also, I can't find any uses of `AssertAllocationAllowed` in our source code, how does kernel32::HeapAlloc interact with it?

Reply:
It can be dead code. Feel free to delete it.

Reply:
Done, lets see if it builds/passes.